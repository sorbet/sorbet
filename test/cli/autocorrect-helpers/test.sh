#!/usr/bin/env bash

tmp="$(mktemp -d)"
infile="test/cli/autocorrect-helpers/autocorrect-helpers.rb"
cp "$infile" "$tmp"

cwd="$(pwd)"
cd "$tmp" || exit 1

if "$cwd/main/sorbet" --silence-dev-message -a autocorrect-helpers.rb 2>&1; then
  echo "Expected to fail!"
  exit 1
fi

echo
echo --------------------------------------------------------------------------
echo

# Also cat the file, to make that the autocorrect applied
cat autocorrect-helpers.rb

rm autocorrect-helpers.rb
rm "$tmp"
