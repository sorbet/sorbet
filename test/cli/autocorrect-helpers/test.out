autocorrect-helpers.rb:4: Method `interface!` does not exist on `T.class_of(S1)` https://srb.help/7003
     4 |  interface!
          ^^^^^^^^^^
  Got `T.class_of(S1)` originating from:
    autocorrect-helpers.rb:3:
     3 |module S1
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:4: Inserted `extend T::Helpers`
     4 |  interface!
        ^

autocorrect-helpers.rb:8: Method `abstract!` does not exist on `T.class_of(S2)` https://srb.help/7003
     8 |  abstract!
          ^^^^^^^^^
  Got `T.class_of(S2)` originating from:
    autocorrect-helpers.rb:7:
     7 |class S2
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:8: Inserted `extend T::Helpers`
     8 |  abstract!
        ^

autocorrect-helpers.rb:12: Method `final!` does not exist on `T.class_of(S3)` https://srb.help/7003
    12 |  final!
          ^^^^^^
  Got `T.class_of(S3)` originating from:
    autocorrect-helpers.rb:11:
    11 |class S3
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:12: Inserted `extend T::Helpers`
    12 |  final!
        ^

autocorrect-helpers.rb:16: Method `sealed!` does not exist on `T.class_of(S4)` https://srb.help/7003
    16 |  sealed!
          ^^^^^^^
  Got `T.class_of(S4)` originating from:
    autocorrect-helpers.rb:15:
    15 |class S4
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:16: Inserted `extend T::Helpers`
    16 |  sealed!
        ^

autocorrect-helpers.rb:20: Method `interface!` does not exist on `T.class_of(S5)` https://srb.help/7003
    20 |  interface!
          ^^^^^^^^^^
  Got `T.class_of(S5)` originating from:
    autocorrect-helpers.rb:19:
    19 |module S5
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:20: Inserted `extend T::Helpers`
    20 |  interface!
        ^

autocorrect-helpers.rb:21: Method `final!` does not exist on `T.class_of(S5)` https://srb.help/7003
    21 |  final!
          ^^^^^^
  Got `T.class_of(S5)` originating from:
    autocorrect-helpers.rb:19:
    19 |module S5
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:20: Inserted `extend T::Helpers`
    20 |  interface!
        ^

autocorrect-helpers.rb:25: Method `abstract!` does not exist on `T.class_of(S6)` https://srb.help/7003
    25 |  abstract!
          ^^^^^^^^^
  Got `T.class_of(S6)` originating from:
    autocorrect-helpers.rb:24:
    24 |class S6
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:25: Inserted `extend T::Helpers`
    25 |  abstract!
        ^

autocorrect-helpers.rb:26: Method `sealed!` does not exist on `T.class_of(S6)` https://srb.help/7003
    26 |  sealed!
          ^^^^^^^
  Got `T.class_of(S6)` originating from:
    autocorrect-helpers.rb:24:
    24 |class S6
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:25: Inserted `extend T::Helpers`
    25 |  abstract!
        ^

autocorrect-helpers.rb:30: Method `mixes_in_class_methods` does not exist on `T.class_of(S7)` https://srb.help/7003
    30 |  mixes_in_class_methods(S1)
          ^^^^^^^^^^^^^^^^^^^^^^
  Got `T.class_of(S7)` originating from:
    autocorrect-helpers.rb:29:
    29 |module S7
        ^
  Autocorrect: Done
    autocorrect-helpers.rb:30: Inserted `extend T::Helpers`
    30 |  mixes_in_class_methods(S1)
        ^
Errors: 9

--------------------------------------------------------------------------

# typed: true

module S1
  extend T::Helpers
  interface!
end

class S2
  extend T::Helpers
  abstract!
end

class S3
  extend T::Helpers
  final!
end

class S4
  extend T::Helpers
  sealed!
end

module S5
  extend T::Helpers
  interface!
  final!
end

class S6
  extend T::Helpers
  abstract!
  sealed!
end

module S7
  extend T::Helpers
  mixes_in_class_methods(S1)
end
